cspice/src/coordinates.rs

```diff
@@ -24,11 +40,11 @@ pub struct AzEl {
 
 impl AzEl {
     /// See [recazl_c](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/recazl_c.html)
-    pub fn from_rect(rect: &Rectangular, azccw: bool, elplsz: bool) -> AzEl {
+    pub fn from_rect(rect: Rectangular, azccw: bool, elplsz: bool) -> Self {
         let mut az_el = AzEl::default();
         spice_unsafe!({
             recazl_c(
-                rect.as_ptr() as *mut SpiceDouble,
+                &*rect as *const SpiceDouble as *mut SpiceDouble,
```

Comment:
I think this should work - just take a mut pointer to the first field in the struct:

```suggestion
    pub fn from_rect(mut rect: Rectangular, azccw: bool, elplsz: bool) -> Self {
        let mut az_el = AzEl::default();
        spice_unsafe!({
            recazl_c(
                &mut t.x as *mut SpiceDouble,
```

Reply:
Oh wow this is so much cleaner! I thought declaring rect as mut would violate the trait bounds but you're totally right. Do you think this is how we should treat other similar situations moving forward? For other struts just align them with repr(C) and pass a mut pointer?